src/Microsoft.AspNet.OData.Shared/Formatter/Serialization/ODataComplexSerializer.cs

```diff
@@ -0,0 +1,128 @@
+﻿// Copyright (c) Microsoft Corporation.  All rights reserved.
+// Licensed under the MIT License.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+#if NETFX // System.Data.Linq.Binary is only supported in the AspNet version.
+using System.Data.Linq;
+#endif
+using System.Diagnostics.Contracts;
+using System.Reflection;
+using System.Xml.Linq;
+using Microsoft.AspNet.OData.Common;
+using Microsoft.OData;
+using Microsoft.OData.Edm;
+
+namespace Microsoft.AspNet.OData.Formatter.Serialization
+{
+    /// <summary>
+    /// Represents an <see cref="ODataSerializer"/> for serializing <see cref="IEdmComplexType" />'s.
+    /// </summary>
+    public class ODataComplexSerializer : ODataEdmTypeSerializer
```

Comment:
>ODataComplexSerializer [](start = 17, length = 22)

That's weird that we should have a complex serializer? 

It should be

Reply:
Its for resourcevalue actually, do you think we should rename?

Reply:
Now I don't have full context on the entirety of this change, but my knee-jerk reaction is "why do we need to reintroduce ODataComplexSerializer"? Before we had `ODataResourceSerializer` we had `ODataEntityTypeSerializer` and `ODataComplexTypeSerializer` which always felt strange because there really isn't any difference between a complex type and an entity type other than the latter has a key defined.

Could we not just use implement `CreateODataValue` on `ODataResourceSerializer`?